module/VuFind/src/VuFind/Related/Bookplate.php

```diff
@@ -0,0 +1,243 @@
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+namespace VuFind\Related;
+
+/**
+ * Related Records: Bookplates
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+class Bookplate implements RelatedInterface
+{
+    /**
+     * Bookplate config
+     */
+    protected $config;
+
+    /**
+     * Data fields (usually Solr)
+     */
+    protected $fields;
+
+    /**
+     * Bookplate strings
+     */
+    protected $bookplateStrs;
+
+    /**
+     * Bookplate image names or full URLs
+     */
+    protected $bookplateImages;
+
+    /**
+     * Bookplate thumbnail image names or thumbnail URLs
+     */
+    protected $bookplateThumbnails;
+
+    /**
+     * URL template for full bookplate
+     */
+    protected $fullUrlTemplate;
+
+    /**
+     * URL temlate for thumbnail
+     */
+    protected $thumbUrlTemplate;
+
+    /**
+     * Display bookplate titles?
+     */
+    protected $displayTitles;
+
+    /**
+     * Constructor
+     *
+     * @param VuFind\Config\PluginManager $configLoader PluginManager
+     */
+    public function __construct(\VuFind\Config\PluginManager $configLoader)
+    {
+        $this->configLoader = $configLoader;
+    }
+
+    /**
+     * Establishes base settings for bookplates.
+     *
+     * @param string                            $settings Settings from config.ini
+     * @param \VuFind\RecordDriver\AbstractBase $driver   Record driver object
+     *
+     * @return void
+     */
+    public function init($settings, $driver)
+    {
+        $config = explode(':', $settings);
+        $configFile = 'config';
+        $configSection = 'Record';
+        if (count($config) == 2) {
+            $configFile = $config[0];
+            $configSection = $config[1];
+        }
+        $this->config = $this->configLoader->get($configFile)->$configSection;
+        $this->fields = $driver->getRawData();
+        $this->bookplateStrs = $this->getBookplateData('titles');
+        $this->bookplateImages = $this->getBookplateData('imgFull');
+        $this->bookplateThumbnails = $this->getBookplateData('imgThumb');
+        $this->fullUrlTemplate = $this->getBookplateFullUrlTemplate();
+        $this->thumbUrlTemplate = $this->getBookplateThumbUrlTemplate();
+        $this->displayTitles = $this->displayBookplateTitles();
+    }
+
+    /**
+     * Get an array of data representing bookplates.
+     *
+     * @param $s string name of data to retrieve.
+     *
+     * @return array
+     */
+    protected function getBookplateData($s)
+    {
+        $data = [
```

Comment:
It seems overcomplicated to build this $data array and then pull exactly one element out of it every time this method is called. Why not just make this `getBookplateData($field)` and then pass in `$this->getBookPlateTitlesField()`, etc., directly from `init()` instead of using these abstract strings? I think that clarifies the code and improves efficiency (since that way we only call each method once, instead of three times).